src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1476,14 +1483,25 @@ internal long SetRequestContent(HttpRequestMessage request, string content)
                 // If Content-Type contains the encoding format (as CharSet), use this encoding format
                 // to encode the Body of the WebRequest sent to the server. Default Encoding format
                 // would be used if Charset is not supplied in the Content-Type property.
-                var mediaTypeHeaderValue = new MediaTypeHeaderValue(ContentType);
-                if (!string.IsNullOrEmpty(mediaTypeHeaderValue.CharSet))
+                try
                 {
-                    try
+                    var mediaTypeHeaderValue = new MediaTypeHeaderValue(ContentType);
+                    if (!string.IsNullOrEmpty(mediaTypeHeaderValue.CharSet))
                     {
                         encoding = Encoding.GetEncoding(mediaTypeHeaderValue.CharSet);
                     }
-                    catch (ArgumentException ex)
+                }
+                catch (FormatException ex)
+                {
+                    if (!SkipHeaderValidation)
+                    {
+                        ErrorRecord er = new ErrorRecord(ex, "WebCmdletContentTypeException", ErrorCategory.InvalidArgument, ContentType);
+                        ThrowTerminatingError(er);
+                    }
```

Comment:
to me.. supplying `-SkipHeaderValidation` means "suppress all header related messages". Perhaps verbose would be better?

Reply:
I guess I see it otherwise; I don't want validation to occur but if it can't be added at all, I believe I would need to know.  For example, I'm aware of cases where header values are needed that do not conform to the standard but the target site requires it; however, if the header cannot even be added, it's something I will need to address because the request is no longer conveying what I expect.

Reply:
Hmm i think I'm a bit confused. We are not discarding any header values.The section of code where this comment thread lives is not the section where headers are added to the request. this section of code is where it tryes to determine request body encoding from `-ContentType`.  We try/catch here because the value `foo` throws `FormatException` on `new MediaTypeHeaderValue("foo")`. 

The section where headers are added, the headers will always be added ([here](https://github.com/markekraus/PowerShell/blob/b5f8245605b39864359577573b4748309525a3df/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs#L1179))

 The only case in which `request,Headers.TryAddWithoutValidation()` or `request.Content.Headers.TryAddWithoutValidation()` would not add headers would be if the headers already existed. This is not possible because we are enumerating a dictionary so the keys will be case-insensitve unique. The headers will always be added and in fact the tests in the project prove that.